### DISCUSSION TITLE
Timestamped per Word through API.

### DISCUSSION THREAD

**Discussion by SpoolifyNL** [openai/whisper#1498] [Q&A]
Hi I'm not the typical programmer. So bear with me. I've mainly used GPT to code stuff for me and obviously I hit roadblocks pretty quick without up to date knowledge in the LLM. My question: I'm using the import openai api to call upon Whisper. 

I've got .wav files with around 13-15 seconds of text. I want to transcribe and mark the time of every word spoken so I can highlight this using another script. Now I'm reading the documentation but I can't figure out how to do this. It's said the openai api Whisper-1 model is functionally the same. But can I even mark the time of every word spoken? Can someone point me to a repo or documentation because there isn't a lot posted about using Whisper with openai API

**Comment by phineas-pta** [comment]
your question should be asked at: https://community.openai.com/tag/whisper

**Comment by Tex2002ans** [comment]
> I want to transcribe and mark the time of every word spoken so I can highlight this using another script. [...] But can I even mark the time of every word spoken?

This functionality is already built into Whisper:

- `--highlight_words`
   - "(requires `--word_timestamps True`) underline each word as it is spoken in srt and vtt (default: `False`)"

If you do a search in previous discussion for that command, you'll see more topics with examples:

- https://github.com/openai/whisper/discussions?discussions_q=is%3Aopen+highlight_words

or search for the keyword "Highlight":

- https://github.com/openai/whisper/discussions?discussions_q=is%3Aopen+highlight

For example, it leads to this fantastic answer by @ryanheise:

- https://github.com/openai/whisper/discussions/1326#discussioncomment-5838635

- - -

Or, see this topic from back in October 2022:

- https://github.com/openai/whisper/discussions/435

where @jianfch put together a few code examples/videos showing off better word-level accuracy.

He even described [how to create the word-highlighted subtitles inside a video using ffmpeg](https://github.com/openai/whisper/discussions/435#discussioncomment-4228825), which seems to be close to what you want.

- - -

**Side Note:** Since that October 2022 thread, Whisper has had quite a few releases and more functionality added. Like now it can output JSON files, with each word marked with exact timings—which may be easier to work with than SRT—depending on what you want to accomplish.

Or a few new commands like:

- `--max_line_width`
    - (requires `--word_timestamps True`) the maximum number of characters in a line before breaking the line (default: `None`)
- `--max_line_count`
    - (requires `--word_timestamps True`) the maximum number of characters in a line before breaking the line (default: `None`)  

which can help generate easier-to-read subtitles.

You can read about that here:

- https://github.com/openai/whisper/pull/1184
- https://github.com/openai/whisper/discussions?discussions_q=is%3Aopen+max_line_width

- - -

> Can someone point me to a repo or documentation because there isn't a lot posted about using Whisper with openai API

In the future, I'd recommend reading this fantastic little article:

- ["The XY Problem" (xyproblem.info)](https://xyproblem.info/)

If you don't explain what ACTUAL PROBLEM you're trying to accomplish, it gets hard.

Good luck with your coding project though as a non-coder... ChatGPT will lead you down lots of roadblocks and completely fabricated commands and fake paths! :P

**Reply by phineas-pta** [reply]
he's using the API, read carefully before spamming

**Reply by Tex2002ans** [reply]
> he's using the API

Simple mistake. I saw they posted in Whisper Discussions, asking about settings that already exist in Whisper.

So, I linked to previous examples/discussion within Whisper's own section + a few newer options which were added in recent versions.

(If someone is completely new, and doesn't follow Whisper's changelogs or discussion very closely, they may not have noticed those topics/settings before.)

Maybe they could now use that information to help in their Whisper (OpenAI API) projects/research too.

Anyway, have a good day. :)
